tests/ast/README.md

```diff
@@ -0,0 +1,38 @@
+## AST Tests
+
+This driver enables testing of the AST generation that will be used in the server-side Snowpark implementation, starting with Phase 0.
+
+All generated AST should be tested using this mechanism. To add a test, create a new file under `tests/ast/data`. Files look like the following example. The test driver sets up the session and looks at the accumulated lazy values in the resulting environment.
+
+N.B. No eager evaluation is permitted, as any intermediate batches will not be observed. This can easily be changed if necessary, however.
+
+```python
+## TEST CASE
+
+df = session.table(tables.table1)
+df = df.filter("STR LIKE '%e%'")
+
+## EXPECTED ENCODED AST
+
+[...]
+
+## EXPECTED UNPARSER OUTPUT
+
+res1 = session.table('table1')
+
+res2 = res1.filter('STR LIKE '%e%'')
```

Comment:
we should update the example here to include the expectation.

Reply:
I added a todo here to revisit and rephrase this file later.